### PR TITLE
[13.0][IMP] queue_job, add decorator job_auto_delay

### LINKED ISSUE #238: [feature] Add `job_auto_delay` decorator
Reminder to move this decorator in queue_job: 

https://github.com/OCA/sale-workflow/blob/713f66478479bd8be77b4b93789bbdfc4c9121cf/sale_automatic_workflow_job/models/automatic_workflow_job.py#L11-L87

### PR DESCRIPTION
**Pull request by kittiu**
Just add a pending TODO from sale_automatic_workflow_job

closes: #238

### CODE REVIEW

**Comment by guewen** [comment]
Thanks, I would appreciate if you keep my authorship though :)

If you have changes to do in the commit, you can add in the commit's body: `Co-authored-by: name <name@example.com>`
As described in https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors

**Comment by kittiu** [comment]
> Thanks, I would appreciate if you keep my authorship though :)
> 
> If you have changes to do in the commit, you can add in the commit's body: `Co-authored-by: name <name@example.com>`
> As described in https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors

Sure! Let me study for it. :)

**Comment by kittiu** [comment]
> Sure! Let me study for it. :)

Done

File: queue_job/job.py [hunk 1]
```diff
@@ -899,3 +899,75 @@ def decorate(func):
...
+    delayed.
+
+    """
+    if func is None:
+        return functools.partial(
+            job_auto_delay, default_channel=default_channel, retry_pattern=retry_pattern
+        )
+
+    def auto_delay(self, *args, **kwargs):
+        if self.env.context.get("job_uuid") or self.env.context.get("_job_force_sync"):
```
Comment (simahawk) [inline queue_job/job.py:953]: maybe better to use the same key as here? https://github.com/OCA/queue/blob/13.0/queue_job/models/base.py#L89

**Review by lmignon — COMMENTED** [review]
@kittiu IMO it's very important to have a unitttest for this new useful functionality

**Comment by guewen** [comment]
@kittiu I proposed an alternative on #270 with tests and a correction in the method + added `test_queue_job_no_delay`

**Pull request closed without merging**